test/Ocean/Hydrostatic_Boussinesq/test_divergence_free.jl

```diff
@@ -0,0 +1,203 @@
+  if bctype == 1
+    ocean_boundary_state!(m, Coastline(), x...)
+  elseif bctype == 2
+    ocean_boundary_state!(m, OceanFloor(), x...)
+  elseif bctype == 3
+    ocean_boundary_state!(m, OceanSurface(), x...)
+  end
+end
+
+struct HomogeneousSimpleBox{T} <: HydrostaticBoussinesqProblem
+  Lˣ::T
+  Lʸ::T
+  H::T
+  τₒ::T
+  fₒ::T
+  β::T
+end
+
+HSBox = HomogeneousSimpleBox
+
+# aux is Filled afer the state
+function ocean_init_aux!(::HBModel, P::HSBox, A, geom)
+  FT = eltype(A)
+  @inbounds y = geom.coord[2]
+
+  Lʸ = P.Lʸ
+  τₒ = P.τₒ
+  fₒ = P.fₒ
+  β  = P.β
+
+  A.τ  = -τₒ * cos(y * 2π / Lʸ)
+  A.f  =  fₒ + β * y
+end
+
+function ocean_init_state!(p::HSBox, state, aux, coords, t)
+  @inbounds z = coords[3]
+  @inbounds H = p.H
+
+  state.u = @SVector [rand(),rand()]
+  state.η = 0
+  state.θ = 20
+end
+
+###################
+# PARAM SELECTION #
+###################
+FT = Float64
+
+const timeend = 86400
+const tout    = 60 * 60
+
+const N  = 4
+const Nˣ = 20
+const Nʸ = 20
+const Nᶻ = 20
+const Lˣ = 4e6
+const Lʸ = 4e6
+const H  = 1000
+
+@show xrange = range(FT(0);  length=Nˣ+1, stop=Lˣ)
+@show yrange = range(FT(0);  length=Nʸ+1, stop=Lʸ)
+@show zrange = range(FT(-H); length=Nᶻ+1, stop=0)
+
+const cʰ = sqrt(grav * H)
+const cᶻ = 0
+
+const τₒ = 1e-1
+const fₒ = 1e-4
+const β  = 1e-11
+
+const αᵀ = 2e-4
+const νʰ = 5e3   # L^2 / t
+const νᶻ = 5e-3
+const κʰ = 1e3
+const κᶻ = 1e-4
+const λʳ = 0 # 2 // 86400
+
+let
+  MPI.Initialized() || MPI.Init()
+  mpicomm = MPI.COMM_WORLD
+
+  ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+  loglevel = ll == "DEBUG" ? Logging.Debug :
+             ll == "WARN"  ? Logging.Warn  :
+             ll == "ERROR" ? Logging.Error : Logging.Info
+  logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+  global_logger(ConsoleLogger(logger_stream, loglevel))
+  @static if haspkg("CUDAnative")
+    device!(MPI.Comm_rank(mpicomm) % length(devices()))
+  end
+
+  @static if haspkg("CuArrays")
+    ArrayType = CuArray
+  else
+    ArrayType = Array
+  end
+
+  brickrange = (xrange, yrange, zrange)
+  topl = StackedBrickTopology(mpicomm, brickrange;
+                              periodicity = (false, false, false),
+                              boundary = ((1, 1), (1, 1), (2, 3)))
+  @show dt = 240 # (Lˣ / cʰ) / Nˣ / N²
+  @show nout = ceil(Int64, tout / dt)
+
+  grid = DiscontinuousSpectralElementGrid(topl,
+                                          FloatType = FT,
+                                          DeviceArray = ArrayType,
+                                          polynomialorder = N,
+                                         )
+
+
+  problem = HSBox{FT}(Lˣ, Lʸ, H, τₒ, fₒ, β)
+
+  model = HBModel{typeof(problem),FT}(problem, cʰ, cʰ, cᶻ, αᵀ, λʳ, νʰ, νᶻ, κʰ, κᶻ)
+
+  dg = OceanDGModel(model,
+                    grid,
+                    Rusanov(),
+                    CentralNumericalFluxDiffusive(),
+                    CentralGradPenalty())
+
+  Q = init_ode_state(dg, FT(0))
```

Comment:
```suggestion
  Q = init_ode_state(dg, FT(0); device=CPU())
```
In PR #483 @simonbyrne added the ability to initialize on the CPU. This should fix the `rand` issue [above](https://github.com/climate-machine/CLIMA/pull/535/files/246fe87ca6c9489e33f6c04775fa62f359a0a4ef#r348186217) I believe.

Reply:
You'll need to add `use GPUifyLoops` to the top of the code as well.

Reply:
Thanks, added these.